### DISCUSSION TITLE
Rapid beam loss and unstable ring in pyAT

### DISCUSSION THREAD

**Discussion by elafmusa** [atcollab/at#590] [Python AT]
Hi everyone,

I would like to address a matter regarding AT which is not clear for me. Specifically, I have applied shift errors to the FCCee lattice while the sextupoles were off. After correcting these errors, I began to gradually ramp up the sextupoles in steps.  I have gathered information on the RMS orbit, tune, beta, and chromaticity parameters after each step.

Here are the parameters before turning on the sextupoles:

rms orbit with sext off : 0.9556868370482995 mu m,  rms orbit y: 1.2284428109798482 mu m
Tune: [0.22406341 0.36018686],  Chromaticity: [ -551.70125889 -2082.3134405 ]
RMS beta beat, x:2.3422637013283185e-08%   y: 1.1510227675206478e-06%
STD beta beat, x:2.341358080244541e-08%   y: 1.1510221970717878e-06%

When i turnd the sextupoles 91% of its strength:

rms orbit with sext off : 0.9560670422001191 mu m,  rms orbit y: 1.2357041576083765 mu m
Tune: [0.22601593 0.359987  ],  Chromaticity: [ -49.90232604 -188.50825473]
RMS beta beat, x:0.2967074434297945%   y: 3.9816937501469125%
STD beta beat, x:0.2650640894226656%   y: 3.9811592444148585%

Upon increasing the strength of the sextupoles to 92%, an error occurred with the message "
ValueError: array must not contain infs or NaNs
". result from the "linopt"

I am a bit confused by the rapid loss of the beam, though the parameter values, such as betas, were not excessively large in the case of the 91% strength.

Thank you in advance

**Comment by swhite2401** [comment]
Dear @elafmusa , sorry I have been busy with IPAC recently and haven't had time to look at your problem. It appears that your lattice becomes unstable but it is difficult to say why with the information you provide.

Few suggestions below:

- What you may try is to track/compute optics in transfer line mode to try to understand what is happening
- It could be that the closed orbit search is failling for some reason, you may also try to give the result of the previous step as input to the next using the optional argument `guess`. Once the closed orbit is found you can provide as input to `get_optics`.
- You may try to re-correct you lattice at each sextupole increase step

@simoneliuzzo has more experience with the FCC lattices, maybe he can think of something obvious that I miss.

**Comment by simoneliuzzo** [comment]
Dear @elafmusa,

I agree with @swhite2401 it is difficult to say like that. I must say that I did only a few tests ramping sextupoles and it worked. I will try as well and let you know if I see the same problem. 
I am puzzled by the fact that with sextupoles at 91/92% you chromaticity is not corrected by 91-92%. May be some sextupole is not ramping?
I agree with @swhite2401 that at each step of the ramp you might need to re-correct the orbit and tunes, although from your data it does not seem necessary (<1um orbit and tunes moved by 1e-3).


P.S. : I see "ramping sextupoles" as an old-school approach, but this is only my opinion. May be it is really needed for FCC.
